app/controllers/api/v1/search_controller.rb

```diff
@@ -207,6 +208,164 @@ def facet_filters
+      end
+
+      swagger_path '/search/bulk_download' do
+        operation :get do
+          key :tags, [
+              'Search'
+          ]
+          key :summary, 'Bulk download study data'
+          key :description, 'Download files in bulk of multiple types from one or more studies via curl'
+          key :operationId, 'search_bulk_download_path'
+          parameter do
+            key :name, :auth_code
+            key :type, :integer
+            key :in, :query
+            key :description, 'User-specific one-time authorization code'
+            key :required, true
+          end
+          parameter do
+            key :name, :accessions
+            key :type, :string
+            key :in, :query
+            key :description, 'Comma-delimited list of Study accessions'
+            key :required, true
+          end
+          parameter do
+            key :name, :file_types
+            key :type, :string
+            key :in, :query
+            key :description, "Comma-delimited list of file types (including 'all' for all files)"
+            key :required, true
+          end
+          response 200 do
+            key :description, 'Curl configuration file with signed URLs for requested data'
+            key :type, :string
+          end
+          response 400 do
+            key :description, 'Invalid study accessions or requested file types'
+          end
+          response 401 do
+            key :description, 'User not signed in'
+          end
+          response 403 do
+            key :description, 'Invalid auth token, or requested download exceeds user download quota'
+            schema do
+              key :title, 'Error'
+              property :message do
+                key :type, :string
+                key :description, 'Error message'
+              end
+            end
+          end
+          response 406 do
+            key :description, 'Accept or Content-Type headers missing or misconfigured'
+          end
+        end
+      end
+
+      def bulk_download
+        totat = params[:auth_code]
+        valid_totat = User.verify_totat(totat)
+        accessions = params[:accessions].split(',').map(&:strip)
+        file_types = params[:file_types].split(',').map(&:strip)
+
+        # sanitize study accessions and file types
+        sanitized_accessions = StudyAccession.sanitize_accessions(accessions)
+        sanitized_file_types = StudyFile::BULK_DOWNLOAD_TYPES & file_types # find array intersection
+
+        # validate request parameters
+        if totat.blank? || valid_totat == false
+          render json: {error: 'Invalid authorization token'}, status: 403 and return
+        elsif sanitized_accessions.blank? || sanitized_file_types.blank?
+          render json: {error: 'Invalid request parameters; study accessions or file types not found'}, status: 400 and return
+        end
+
+        # load the user from the auth token
+        requested_user = valid_totat
+
```

Comment:
I think everything below here belongs in a static method or, better yet, a new BulkDownloadService class in app/lib

Reply:
f5a85e5